### PR TITLE
sed: enable more platforms

### PR DESCRIPTION
**Pull request by fingolfin**
Also, this is the first rebuild in 2 years, so it also gets us JLLWrapper for this JLL

### CODE REVIEW

**Comment by fingolfin** [comment]
Hmmm, Windows builds fail:
```
[00:25:31] sed/sed-compile.o: In function `sprintf':
[00:25:31] /opt/x86_64-w64-mingw32/x86_64-w64-mingw32/sys-root/include/stdio.h:366: undefined reference to `__chk_fail'
[00:25:31] sed/sed-execute.o: In function `do_list':
[00:25:31] /opt/x86_64-w64-mingw32/x86_64-w64-mingw32/sys-root/include/stdio.h:366: undefined reference to `__chk_fail'
...
```

**Comment by fingolfin** [comment]
Will try this from `Opus_jll`:
```
elif [[ "${target}" == *-mingw* ]]; then
    # Fix error
    #     /opt/x86_64-w64-mingw32/bin/../lib/gcc/x86_64-w64-mingw32/8.1.0/../../../../x86_64-w64-mingw32/bin/ld: src/opus_compare.o: in function `fread':
    #     /opt/x86_64-w64-mingw32/x86_64-w64-mingw32/sys-root/include/stdio.h:812: undefined reference to `__chk_fail'
    # See https://github.com/msys2/MINGW-packages/issues/5868#issuecomment-544107564
    export LDFLAGS="-lssp"
fi
```

**Comment by giordano** [comment]
```
[00:49:44] cc -std=gnu11    -g -O2  -lssp -o sed/sed.exe sed/sed-compile.o sed/sed-debug.o sed/sed-execute.o sed/sed-mbcs.o sed/sed-regexp.o sed/sed-sed.o sed/sed-utils.o sed/libver.a lib/libsed.a    
[00:49:44] sed/sed-compile.o: In function `sprintf':
[00:49:44] /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/stdio.h:366: undefined reference to `__chk_fail'
[00:49:44] sed/sed-execute.o: In function `sprintf':
[00:49:44] /opt/i686-w64-mingw32/i686-w64-mingw32/sys-root/include/stdio.h:366: undefined reference to `__chk_fail'
 ```
I guess they placed `LDFLAGS` too early in the compiler invocation 😬

File: S/sed/build_tarballs.jl [hunk 1]
```diff
@@ -12,15 +12,23 @@ sources = [
...
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/sed-4.8/
+cd $WORKSPACE/srcdir/sed-*
+if [[ "${target}" == *-mingw* ]]; then
+    # Fix error
+    #    sed/sed-compile.o: In function `sprintf':
+    #    /opt/x86_64-w64-mingw32/x86_64-w64-mingw32/sys-root/include/stdio.h:366: undefined reference to `__chk_fail'
+    # See https://github.com/msys2/MINGW-packages/issues/5868#issuecomment-544107564
+    export LDFLAGS="-lssp"
```
Comment (benlorenz) [inline S/sed/build_tarballs.jl:21]:
```suggestion
    export LIBS="-lssp"
```
`LDFLAGS`: Don't use this variable to pass library names (-l) to the linker; use LIBS instead.

**Pull request merged**